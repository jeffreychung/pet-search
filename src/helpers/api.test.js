import {
  getWhereClause,
  getQuery,
} from './api';

describe('api', () => {
  describe('getWhereClause', () => {
    it('should format a single querystring', () => {
      const searches = {
        search_1: {
          id: 'search_1',
          searchField: 'animal_type',
          searchCondition: 'is',
          searchValue: 'Livestock',
        },
      };

      expect(getWhereClause(searches)).toMatchSnapshot();
    });

    it('should format multiple querystring', () => {
      const searches = {
        search_1: {
          id: 'search_1',
          searchField: 'animal_type',
          searchCondition: 'is',
          searchValue: 'Livestock',
        },
        search_2: {
          id: 'search_1',
          searchField: 'animal_type',
          searchCondition: 'is_not',
          searchValue: 'Bird',
        },
      };

      expect(getWhereClause(searches)).toMatchSnapshot();
    });
  });

  describe('getQuery', () => {
    it('format a query', () => {
      const searches = {
        search_1: {
          id: 'search_1',
          searchField: 'animal_type',
          searchCondition: 'is',
          searchValue: 'Livestock',
        },
        search_2: {
          id: 'search_1',
          searchField: 'animal_type',
          searchCondition: 'is_not',
          searchValue: 'Bird',
        },
      };
      expect(getQuery(searches)).toMatchSnapshot();
    });
  });
});
