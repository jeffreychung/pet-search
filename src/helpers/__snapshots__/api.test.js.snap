exports[`api getQuery format a query 1`] = `
"SELECT *
    WHERE animal_type = \'Livestock\' AND animal_type != \'Bird\'
    ORDER BY datetime
    LIMIT 10
    OFFSET 6
  "
`;

exports[`api getWhereClause should format a single querystring 1`] = `"WHERE animal_type = \'Livestock\'"`;

exports[`api getWhereClause should format multiple querystring 1`] = `"WHERE animal_type = \'Livestock\' AND animal_type != \'Bird\'"`;

exports[`api getWhereClause should skip null for one search 1`] = `""`;

exports[`api getWhereClause should skip null two searches 1`] = `"WHERE animal_type != \'Bird\'"`;
